src/main/java/com/opentable/metrics/graphite/GraphiteSenderWrapper.java

```diff
@@ -44,7 +43,12 @@
      * metrics.
      *
      * <p>
-     * This is an unfortunate hack, and it would be nice to be able to remove it.
+     * As of DropWizard 3.1.3+, this is PROBABLY not needed anymore, but the wrapper is maintained for providing delegation purposes
+     * and for tracking connectionFailures and closes for diagnostics.
+     * (In DropWizard 3.1.3+, each reporting period, a new socket (including DNS resolution) is created, metrics are sent, and
+     * the socket is closed. All the previous issues occurred because of atttempts to reuse a socket)
+     * Hence, this class is now mostly a "dumb" delegator, with little extra logic other than
+     * metrics wrapping, and a restart per hour.
```

Comment:
I think we should dump the hourly restart.

Reply:
It's out of an excess of caution tbh. I could be argued/cajoled into it ;)